core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/Processor.scala

```diff
@@ -366,7 +366,7 @@ class Processor(var dataProcessor: IOperatorExecutor, val tag: WorkerTag) extend
   override def completed: Receive =
     disallowDataMessages orElse disallowUpdateInputLinking orElse super.completed
 
-  private[this] def beforeProcessingBatch(): Unit = {
+  protected[this] def beforeProcessingBatch(): Unit = {
```

Comment:
I assume these changes are also not needed as well?

Reply:
yes